docs/guide/nodes-for-xod-in-cpp/README.md

```diff
@@ -0,0 +1,135 @@
+
+When node’s pins meaning is utterly apparent, we recommend omitting pin labels
+to reduce visual clutter. In our case the meaning of output is obvious, but the
+input is not so simple. The angle could be in radians or degrees. For this
+example, we’re going to choose [radians](https://en.wikipedia.org/wiki/Radian)
+because they are often more convenient in mathematical computations and they are
+SI units. To make it clear, let’s give label `RAD` to the input terminal.
+
+![Cosine terminals](./step1.patch.png)
+
+<div class="ui segment note">
+<span class="ui ribbon label">Note</span>
+Despite the recommendation, many standard XOD nodes include labels even in
+trivial cases. It is legacy; we’ll fix them in future.
+</div>
+
+## Declare it’s a C\++ node
+
+The next step is placing `not-implemented-in-xod` node from `xod/patch-nodes`.
+It’s a dummy node that tells XOD: “Ignore all regular nodes on this patch, I
+have a C\++ implementation, use it.”
+
+![not-implemented-in-xod node](./step2.patch.png)
+
+Double click the `not-implemented-in-xod` node to drill-down. You’ll see an
+embedded code editor with a template for a start.
+
+## Write down the code
+
+In most cases, when you implement a C\++ node you want to read values from
+inputs, do something useful with them, and put some values into outputs. To
+communicate with inputs and outputs, XOD provides C\++ functions `getValue` and
+`emitValue`. They’re better demonstrated with an example of complete `cos`
+implementation:
+
+```cpp
+// Ignore this for now. The definition is required by XOD but we
+// not use it for cos
+struct State { };
+
+// This is an obligatory marker. XOD will put pin definitions there
+\{{ GENERATED_CODE }}
+
+// The `evaluate` function is the node’s entry point. It is the only
+// function XOD requires to implement. The context parameter is a
+// black-box object you’ll pass to various API functions.
+void evaluate(Context ctx) {
+    // `getValue` function reads a pin. In angle brackets, it takes a
+    // name of the pin in form input_PINLABEL (input_RAD in our case)
+    Number x = getValue<input_RAD>(ctx);
+
+    // Next, we use regular C++ to perform some actions. Our case is trivial.
+    // All we have to do is to call the standard C++ cos function.
+    // Note the data type `Number`. It’s the type XOD uses to represent numbers
+    // on the current platform.
+    Number result = cos(x);
+
+    // `emitValue` is like `getValue`, but it writes values rather than read.
+    // Note we use `OUT` label to access our unlabeled output terminal.
+    // `OUT` is the default name when you omit the label.
+    emitValue<output_OUT>(ctx, result);
+}
+```
+
+You may wonder why we used `output_OUT` to refer our unlabeled output terminal.
+That’s a default name when you omit the label. Likewise, if you omit a label for
+an input, use `input_IN` to refer to it. And if you have got several unnamed
+terminals use `input_IN1`, `input_IN2`, `output_OUT1`, `output_OUT2`, etc.
```

Comment:
Maybe will be useful to add a hint about hovering pin in the Inspector to get these names or just count it from left to right and then from the top down.

Reply:
Done